### PR TITLE
Fixes for MS Visual Studio 2017

### PR DESCRIPTION
**Pull request by OgreTransporter**
I try to compile texstudio on Windows with MS VS2017 and run into a few problems. These comits are fixing most errors.

### CODE REVIEW

**Comment by sunderme** [comment]
you should post a tutorial how to compile txs with visual studio...

**Comment by OgreTransporter** [comment]
I can do that, but I have to make a few other changes before ;-)

File: src/configdialog.cpp [hunk 1]
```diff
@@ -244,7 +244,7 @@ void ShortcutDelegate::setModelData(QWidget *editor, QAbstractItemModel *model,
 				if (UtilsUi::txsConfirmWarning(QString(ConfigDialog::tr("The shortcut <%1> is already assigned to the command:")).arg(value) + "\n" + duplicate + "\n\n" + ConfigDialog::tr("Do you wish to remove the old assignment and bind the shortcut to the new command?"))) {
 					//model->setData(mil[0],"",Qt::DisplayRole);
 					foreach (QTreeWidgetItem *twi, li) {
-						if (twi and twi->text(2) == value) twi->setText(2, "");
+                        if (twi && twi->text(2) == value) twi->setText(2, "");
```
Comment (timhoffm) [inline src/configdialog.cpp]: Can you please check the code formatting settings of your editor? We're using tabs for indentation and your edits replace them by 4 spaces.

**Comment by timhoffm** [comment]
Thanks. Looks mostly good.

File: src/configdialog.cpp (hunk 1 above)
Comment (OgreTransporter) [reply src/configdialog.cpp]: I'm sorry about that! Fix: #429

**Pull request merged**